### PR TITLE
Basic Apache Age connection in Swift

### PR DESCRIPTION
**Pull request by FahadZaheerfzr**
This pull request contains the initial connection of PostgreSQL in Swift

### CODE REVIEW

File: drivers/age-swift/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# age-swift
```
Comment (jasperblues) [inline drivers/age-swift/README.md:1]: Perhaps you can add a quick start guide (Swift usage), with links to AGE quick start (for setting up db).

File: drivers/age-swift/Sources/age-swift/age.swift [hunk 2]
```diff
@@ -0,0 +1,108 @@
+//
```
Comment (jasperblues) [inline drivers/age-swift/Sources/age-swift/age.swift:1]: Is there a standard AGE copyright header, including Apache license details?

File: drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import XCTest
+@testable import age_swift
+
+let host = "127.0.0.1";
+let port = 5432 ;
+let user = "root";
+let db_name = "demo";
+
+final class age_swiftTests: XCTestCase {
+    func testExample() throws {
```
Comment (jasperblues) [inline drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift:10]: Suggest to name the test method as following: 

```scenarioName_should_expectedOutcome()```

For example: 

```connectingWithSuppliedParams_shouldNotError()``` 

Suggest to also add some more scenarios: 

* What happens when the incorrect params are supplied? What's the expected error message? Ensure it does not crash. 
* What happens when we execute CYPHER statements returning vertices? 
* What happens when we execute CYPHER statements returning vertices and edges?

File: drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        // This is an example of a functional test case.
+        // Use XCTAssert and related functions to verify your tests produce the correct
+        // results.
+        let age = Age()
+
+        let connection_param:[String:Any] = ["host":host,"port":port, "user":user, "dbname":db_name]
```
Comment (jasperblues) [inline drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift:16]: How about having a type to represent the connection params? 

So that the IDE (Xcode) can code-complete these? Rather than the user having to remember the name of strings, and perhaps type them incorrectly?

File: drivers/age-swift/README.md (hunk 1 above)
Comment (FahadZaheerfzr) [reply drivers/age-swift/README.md:1]: Alright I will do that in my next pull request

File: drivers/age-swift/Sources/age-swift/age.swift (hunk 2 above)
Comment (FahadZaheerfzr) [reply drivers/age-swift/Sources/age-swift/age.swift:1]: Will add this too

File: drivers/age-swift/README.md (hunk 1 above)
Comment (jasperblues) [reply drivers/age-swift/README.md:1]: Once the readme is set up, I will comment on the usage and perhaps have some suggestions for the correct level of abstraction for a driver - it should be easy to use in the most common scenarios, possible in the edge cases. 

Great work so far!

File: drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift (hunk 3 above)
Comment (FahadZaheerfzr) [reply drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift:10]: Yes will keep adding new test cases, in my upcoming pull requests

File: drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift (hunk 4 above)
Comment (FahadZaheerfzr) [reply drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift:16]: Makes sense. Will try to do this too in my upcoming pull request

File: drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift (hunk 3 above)
Comment (jasperblues) [reply drivers/age-swift/Tests/age-swiftTests/age_swiftTests.swift:10]: You can add more commits to this PR. They can be squashed down when merging.

**Comment by dehowef** [comment]
Hey Fahad, thank you for your contributions. I think something that would be helpful is if you squash your commits down. This way your pull request will be more concise with what you are adding to the project. Also, adding a short description detailing the changes under your commit for what changes were added is helpful for helping other contributors to know what you are adding with each commit. You can reference commits done on the main branches for example. Thank you !

**Review by dehowef — CHANGES_REQUESTED** [review]
Hey Fahad, thank you for your contributions. I think something that would be helpful is if you squash your commits down. This way your pull request will be more concise with what you are adding to the project. Also, adding a short description detailing the changes under your commit for what changes were added is helpful for helping other contributors to know what you are adding with each commit. You can reference commits done on the main branches for example. Thank you !

**Comment by jasperblues** [comment]
image

@dehowef With Github its also possible to squash while merging.
